### PR TITLE
Use mongo-java-driver 3.0.0

### PR DESCRIPTION
**Pull request by kfowler**
https://github.com/devbliss/mongojack/issues/103

Minimal changes to get mongojack to compile with mongo-java-driver 3.0.0.
 
 - remove deprecated methods that use APIs not present in 3.0.0
  - depend on flapdoodle 1.47.4-SNAPSHOT w/ mongo-java-driver 3.0.0
  support
  - use mongo-java-driver 3.0.0 in dependencies with provided scope
  - bump version number to 2.3.2-SNAPSHOT

All tests pass.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,7 +25,8 @@
         <dependency>
             <groupId>org.mongodb</groupId>
             <artifactId>mongo-java-driver</artifactId>
-            <version>2.13.0</version>
+            <version>3.0.0</version>
+            <scope>provided</scope>
```
Comment (benmccann) [inline pom.xml]: why provided? so that you can use 2.13 or 3.0? is this meant to be compatible with both?

File: pom.xml [hunk 2]
```diff
@@ -56,7 +57,7 @@
         <dependency>
             <groupId>de.flapdoodle.embed</groupId>
             <artifactId>de.flapdoodle.embed.mongo</artifactId>
-            <version>1.47.0</version>
+            <version>1.47.4-SNAPSHOT</version>
```
Comment (benmccann) [inline pom.xml]: we should wait to merge this until we can rely on a non-snapshot version

File: src/main/java/org/mongojack/MapReduce.java [hunk 3]
```diff
@@ -250,11 +250,6 @@ private MapReduceCommand(String map, String reduce,
                 command.setScope(scope);
             }
             command.setVerbose(verbose);
-            if (extra != null) {
```
Comment (benmccann) [inline src/main/java/org/mongojack/MapReduce.java]: why remove this?

File: pom.xml (hunk 1 above)
Comment (kfowler) [inline pom.xml]: Yes: provided so that projects using either 2.x or 3.0 can depend on one release of mongojack.

File: pom.xml (hunk 2 above)
Comment (kfowler) [inline pom.xml]: Definitely. 1.47.4 looks like it will support mongo-java-driver 3.0.0

File: src/main/java/org/mongojack/MapReduce.java (hunk 3 above)
Comment (kfowler) [inline src/main/java/org/mongojack/MapReduce.java]: com.mongodb.MapReduceCommand.addExtraOption() is removed in 3.0.0. was deprecated in 2.13.x

Comment (benmccann) [inline src/main/java/org/mongojack/MapReduce.java]: We should probably also remove the variable `extra` then or else there's an option around that doesn't do anything...

File: pom.xml (hunk 1 above)
Comment (benmccann) [inline pom.xml]: I'd probably remove the provided scope. A project can still exclude the driver and pull in 2.x if they want, but I'm not sure we're super interested in maintaining it such that it works on both

**Pull request merged**